osu.Game/Screens/Play/HUD/MultiplayerGameplayLeaderboard.cs

```diff
@@ -55,9 +67,14 @@ private void load(OsuConfigManager config, IAPIProvider api)
 
             foreach (var userId in playingUsers)
             {
-                var trackedUser = CreateUserData(userId, scoreProcessor);
+                var user = multiplayerClient.Room?.Users.FirstOrDefault(u => u.UserID == userId);
```

Comment:
This is a little dangerous because the list is modified on the update thread. For example, if a user disconnects/exits right as load starts, then this may/will cause an enumeration exception...

Is it possible to change the ctor to `MultiplayerUser[]` instead, or is that too large of a change?

Reply:
I initially tried to do this, but it didn't seem to work initially. Will test one more to make sure it can't be easily done.

Reply:
Otherwise, could this be nullable and set from the scheduled callback of the `userLookupCache` call below?

Idk, I can merge this but it feels sketchy... This whole class probably needs a refactor.